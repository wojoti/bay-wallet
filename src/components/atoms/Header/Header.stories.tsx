import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import Header from './Header';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Atoms/Header',
  component: Header,
  argTypes: {
    color: {
      control: {type: 'color'},
    },
    align: {
      control: {type: 'select'},
      options: ['left', 'auto', 'right', 'center', 'justify', undefined],
    },
  },
} as ComponentMeta<typeof Header>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Header> = args => <Header {...args} />;

export const Example = Template.bind({});
Example.args = {
  children: 'BayWallet',
  color: 'black',
  size: 34,
  testId: '',
  align: 'left',
};

export const CenteredExample = Template.bind({});
CenteredExample.args = {
  children: 'New cards in few clicks',
  color: '#6B6B6B',
  size: 40,
  testId: '',
  align: 'center',
};
