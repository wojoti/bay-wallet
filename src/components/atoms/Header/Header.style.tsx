import {StyleSheet} from 'react-native';
import {boldFont} from 'style/style';

interface Props {
  color?: string;
  size?: number;
  align?: 'left' | 'auto' | 'right' | 'center' | 'justify' | undefined;
}
const style = ({color, size, align}: Props) =>
  StyleSheet.create({
    wrapper: {
      margin: 0,
      padding: 0,
      color: color || 'black',
      textAlign: align || 'left',
      fontSize: size || 24,
      fontFamily: boldFont,
    },
  });
export default style;
