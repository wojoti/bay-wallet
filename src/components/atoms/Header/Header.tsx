import {PropsWithChildren} from 'react';
import {Text} from 'react-native';
import style from './Header.style';

export type HeaderProps = PropsWithChildren<{
  color?: string;
  testId?: string;
  align?: 'left' | 'auto' | 'right' | 'center' | 'justify' | undefined;
  size?: number;
}>;

const Header = ({color, align, size, testId, children}: HeaderProps) => {
  return (
    <Text style={style({color, size, align}).wrapper} testID={testId}>
      {children}
    </Text>
  );
};

export default Header;
