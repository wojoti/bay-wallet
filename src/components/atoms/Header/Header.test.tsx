import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Header, {HeaderProps} from './Header';

const props: HeaderProps = {
  testId: 'test-header-id',
};

const modifiedProps: HeaderProps = {
  ...props,
  color: 'red',
  size: 100,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(
    <Header {...props}>test header</Header>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render header', () => {
  render(<Header {...props}>test header</Header>);
  const headerElement = screen.getByTestId('test-header-id');
  expect(headerElement).toBeOnTheScreen();
});

test('should render header with text', () => {
  render(<Header {...props}>test header</Header>);
  const headerElement = screen.getByTestId('test-header-id');
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('test header');
});

test('should render header with modified style through props', () => {
  render(<Header {...modifiedProps} />);
  const headerElement = screen.getByTestId('test-header-id');
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveStyle({
    color: 'red',
    fontSize: 100,
  });
});
