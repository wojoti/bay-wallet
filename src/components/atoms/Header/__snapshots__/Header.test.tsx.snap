// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Text
  style={
    {
      "color": "black",
      "fontFamily": "Poppins-Bold",
      "fontSize": 24,
      "letterSpacing": undefined,
      "textAlign": "left",
    }
  }
  testID="test-header-id"
>
  test header
</Text>
`;
